src/plugins/event-plugins/JsErrorPlugin.ts

```diff
@@ -62,10 +65,15 @@ export class JsErrorPlugin implements Plugin {
     }
 
     private eventHandler = (errorEvent: ErrorEvent) => {
```

Comment:
issue: I think we want to add the filter to `promiseRejectEventHandler` separately, so that the filter will get the complete event in both cases. We'll also need to avoid calling `eventHandler` from `promiseRejectEventHandler` so the filter doesn't get applied twice (or something to that effect).

Reply:
Similarly, for `record`, we don't want the error to pass through the filter.

I suggest doing an extract method refactor on `eventHandler` so that the three functions still share common code, but, the original `eventHandler` only does the `filter` check.

Reply:
Made the following changes:
1. Pull out the actual `event` creating activity into `recordJsErrorEvent(error: any)` method. This method simply records the error event, and is used across `record(), promiseRejectEventHandler(), eventHandler()`.
2. `record()` will directly invoke `recordJsErrorEvent` instead of going through `eventHandler` to avoid going through filtering logic.
3. `promiseRejectEventHandler` will now do its own filtering and call `recordJsErrorEvent` if the error should be recorded